src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
 
 .. contents:: Table of Contents
 
+Preliminary Steps
+-----------------
+
+The following documentation chapters expect a running Ceph Cluster and at
+least a running `dashboard` manager module (with few exceptions). This
+chapter gives an introduction on how to set up such a system for development,
+without the need to set up a full blown production environment. All in this
+chapter introduced options are based on a so called `vstart` environment. 
+
+.. note::
+
+  Every vstart environment needs Ceph `to be compiled`_ from its Github
+  repository, though Docker environments simplify that step by providing a
+  shell script that contains those instructions.
+
+.. _to be compiled: https://docs.ceph.com/docs/master/install/build-ceph/
+
+vstart
+~~~~~~
+
+"vstart" is actually a shell script in the `src/` directory of the Ceph
+repository (``src/vstart.sh``). It is used to start a single node Ceph
+cluster on the machine where it is executed. Several required and some
+optional Ceph internal services are started automatically when it is used to
+start a Ceph cluster. `vstart` is the basis for the three most commonly used
+development environments in Ceph Dashboard.
+
+Host vs Docker
+~~~~~~~~~~~~~~
+
+This document introduces you to three different development environments, all
+based on vstart. Those are:
+
+- vstart running on your host system
+
+- vstart running in a Docker environment
+
+  * ceph-dev-docker_ (mostly used by SUSE employees)
+  * ceph-dev_ (mostly used by RedHat employees)
+
+.. note::
+
+  Independently of which of these environments you will choose, you need to
+  compile on Ceph in that environment. If you compiled Ceph on your host
+  system, you would have to recompile it on Docker to be able to switch to a
+  Docker based solution. The same is true vice versa. If you previously used
+  a Docker development environment and compiled Ceph there and you now want
+  to switch to your host system, you will also need to recompile Ceph (or
+  compile Ceph using another separate repository).
+
+Development environment on your host system
```

Comment:
If it really is pollution is a question of perspective. I have the intention to (also) work on my host system and would never consider this to be polluting my host. I'd hence rather try to stay neutral here and not put it that way

Reply:
Regardless, I suggest to include a remark that setting up a development environment on the host system will result in a large number of additional packages to be installed, which requires additional disk space.

Reply:
Well, it might make sense to note that additional packages need to be installed on the host system, though, I'd like to point out that the disk space of the root file system will in each case be reduced by the size of the dependencies which are required to compile Ceph, regardless of which development environment is chosen, host or container. Hence I'm inclined to put this information on neither side of the comparison, if you agree.

When I thought about this topic, it occurred to me that the disk space used by the `build/` directory after compilation might be more noteworthy. I'll find a place to mention that.

Reply:
Big caveat of this approach is that you may need to upgrade you host OS to be able to catch up with latest Ceph build deps (it happened to me when I joined: I was running Fedora 27 and all of a sudden Ceph switched GCC to v8 so I had to migrate to Fedora 28, and that caused lot of side-effects with other tools... That led me to choose the containerized approach over the bare-metal one).